doc/source/text.rst

```diff
@@ -306,23 +306,30 @@ The same alignment can be used when ``others`` is a ``DataFrame``:
 Concatenating a Series and many objects into a Series
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-All one-dimensional list-likes can be combined in a list-like container (including iterators, ``dict``-views, etc.):
+Several items can be combined a list-like container (including iterators, ``dict``-views, etc.), which may contain ``Series``, ``Index`` and ``np.ndarray``.
+Note that ``Index`` will align as well, so for the purpose of this example, we change ``s`` and ``u`` from above to also have a string index:
 
 .. ipython:: python
 
-    s
-    u
-    s.str.cat([u.values,
-               u.index.astype(str).values], na_rep='-')
+    s_values = np.array(['a', 'b', 'c', 'd'], dtype=object)
+    s2 = pd.Series(s_values, index=s_values)
+    s2
```

Comment:
add this in multiple blocks as its too much to complicated here

Reply:
this whole example needs to be simpler. maybe just leave the index as integers to avoid confusion, IOW focus less on the join in str.cat and more on the list-lke things that are going on.

Reply:
The thing this example tries to show is that it works for `Series`, `ndarray`, and `Index`, and `Index` can only be aligned with a non-integer index. This whole example would be so so much easier with #22225. I've opened an issue for the DF discussion a while ago, and asked you to comment #24046.